app/controllers/user/registrations_controller.rb

```diff
@@ -0,0 +1,45 @@
+class User::RegistrationsController < Devise::RegistrationsController
+  before_action :configure_sign_up_params, only: :create
+
+  def create
+    if params[:flag]
+      build_resource(user_params)
+      resource.osbb_attributes = osbb_params
+      if resource.valid?
+        resource.save
+        yield resource if block_given?
+        if resource.persisted?
+          if resource.active_for_authentication?
+            set_flash_message! :notice, :signed_up
+            sign_up(resource_name, resource)
+            respond_with resource, location: after_sign_up_path_for(resource)
+          else
+            set_flash_message! :notice, :"signed_up_but_#{resource.inactive_message}"
+            expire_data_after_sign_in!
+            respond_with resource, location: after_inactive_sign_up_path_for(resource)
+          end
+        end
+      else
+        render :new
+        clean_up_passwords resource
+        set_minimum_password_length
+      end
+    else
+      super
+    end
+  end
+
+  protected
+
+  def osbb_params
+    params[:user][:osbb_attributes].permit(:name, :phone, :email, :website)
+  end
+
+  def user_params
+    sign_up_params.merge({ role: User::LEAD })
```

Comment:
this role should only be applied if osbb is newly created,
as it is now looks like every new user will be a lead even without osbb attributes

Reply:
refactor this